File: src/main/java/szewek/mcflux/fluxcompat/LazyEnergy.java
package szewek.mcflux.fluxcompat;

import szewek.fl.energy.IEnergy;

public final class LazyEnergy extends ForgeEnergyCapable implements FluxCompat.Convert {
	boolean notEnergy = false;
	IEnergy ie;

	@Override
	public boolean canInputEnergy() {
		return ie != null && ie.canInputEnergy();
	}

	@Override
	public boolean canOutputEnergy() {
		return ie != null && ie.canOutputEnergy();
	}

	@Override
	public long inputEnergy(long amount, boolean sim) {
		return ie != null ? ie.inputEnergy(amount, sim) : 0;
	}

	@Override
	public long outputEnergy(long amount, boolean sim) {
		return ie != null ? ie.outputEnergy(amount, sim) : 0;
	}

	@Override
	public long getEnergy() {
		return ie != null ? ie.getEnergy() : 0;
	}

	@Override
	public long getEnergyCapacity() {
		return ie != null ? ie.getEnergyCapacity() : 0;
	}

	public boolean isReady() {
		return ie != null;
	}

	@Override
	public EnergyType getEnergyType() {
		return notEnergy ? EnergyType.NONE : EnergyType.LAZY;
	}
}

File: src/main/java/szewek/mcflux/fluxcompat/LazyEnergy.kt
package szewek.mcflux.fluxcompat

import szewek.fl.energy.IEnergy

class LazyEnergy : ForgeEnergyCapable(), FluxCompat.Convert {
	internal var notEnergy = false
	internal var ie: IEnergy? = null

	val isReady: Boolean get() = ie != null

	override fun canInputEnergy() = ie != null && ie!!.canInputEnergy()
	override fun canOutputEnergy() = ie != null && ie!!.canOutputEnergy()
	override fun inputEnergy(amount: Long, sim: Boolean) = ie?.inputEnergy(amount, sim) ?: 0
	override fun outputEnergy(amount: Long, sim: Boolean) = ie?.outputEnergy(amount, sim) ?: 0
	override fun getEnergy() = ie?.energy ?: 0
	override fun getEnergyCapacity() = ie?.energyCapacity ?: 0

	override val energyType get() = if (notEnergy) EnergyType.NONE else EnergyType.LAZY
}
